tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/coprocessor/TransactionProcessor.java

```diff
@@ -356,8 +356,8 @@ public InternalScanner preCompactScannerOpen(ObserverContext<RegionCoprocessorEn
   @Override
   public void postCompact(ObserverContext<RegionCoprocessorEnvironment> e, Store store, StoreFile resultFile,
                           CompactionRequest request) throws IOException {
-    // Persist the compaction state after a succesful compaction
-    if (compactionState != null) {
+    // Persist the compaction state after a successful compaction
+    if (Boolean.TRUE.equals(pruneEnable)) {
```

Comment:
It would be better to change this check to `compactionState != null`. Otherwise there is a race condition during `initializePruneState()`, where `pruneEnable` could be `true`, but `compactionState` is still not initialized.